### PR TITLE
ENG-88: Goal oriented testing implementation

### PR DESCRIPTION
**Pull request by Iamdavidonuh**
This PR introduces goal-oriented testing, enhancing our feedback loop for validation. With this approach, we can efficiently validate features, debug issues, and create test cases to cover edge cases—all locally.

Key Additions

- [x]  Query-Based Testing – Loads Parquet files into DuckDB’s Rust crate as an in-memory database, enabling real SQL queries for validation and debugging.

- [x] Dataset Processing Tracking – Accurately monitors and tracks the number of datasets processed in a pipeline run.

- [x] Optimize `ProcessWatcher's` `gather_parent_process` not to panic if no parent_id is found

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,12 +26,12 @@ setup_nextest:
 	@which cargo-nextest >/dev/null || cargo install cargo-nextest
 
 test: setup_nextest ## Run tests single threaded due to s3 race conditions
-	AWS_REGION=us-east-2 \
```
Comment (MicheleVerriello) [inline Makefile:29]: This part allows to perform tests on real aws environment. i think we should keep it.

**Review by MicheleVerriello — APPROVED** [review]
Seems fine, good job!

**Comment by davincios** [comment]
Locally there might be an issue:
image

**Comment by davincios** [comment]
Yes I tried to fix the local issue for quite sometime but couldn't get it to work also tried to merge staging into the current branch but got issues with the EC2 pricing changes.

**Review by davincios — APPROVED** [review]
